### ISSUE TITLE
Spec-first development with `openapi-generator`?

### ISSUE THREAD

**Issue by sherifnada** [tiangolo/fastapi#5557]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code
I have the following OpenAPI spec
```yaml
openapi: 3.0.0
info:
  description: |
    Connector Builder Server API

  version: "1.0.0"
  title: Connector Builder Server API
  contact:
    email: myemail@gmail.com
  license:
    name: MIT
    url: "https://opensource.org/licenses/MIT"

paths:
  /hello_world:
    get:
      description: "Sample endpoint"
      responses:
        "200":
          description: Successful operation
          content:
            application/json:
              schema:
                type: object
                properties:
                  hello:
                    type: string
```

When I run `openapi-generator -i <path_to_spec> -g python-fastapi`, a file `src/openapi/apis/default_api.py` (among other things) is autogenerated containing the following code:

```python
router = APIRouter()

@router.get(
    "/hello_world",
    responses={
        200: {"model": InlineResponse200, "description": "Successful operation"},
    },
    tags=["default"],
)
async def hello_world_get(
) -> InlineResponse200:
    """Sample endpoint"""
    ...
```

### My Question
How is a developer using this generator expected to implement and maintain these endpoints on an ongoing basis as the spec evolves over time? In other words, is there any way to: 
1. override the behavior of these methods from outside this file (so my implementation logic doesn't get overwritten when I regenerate the file) , and
2. detect if an API method in the spec was not implemented by the developer, and raise an exception during CI or tests

If this was say, Java, the endpoints would be encapsulated under an interface like: 

```java
// DefaultAPI.java
// This interface is autogenerated 
interface DefaultAPI() {
  InlineResponse200 helloWorld();
}
```

Then in a separate file I'd implement the API 
```java
// DefaultApiImplementation.java
class DefaultApiImplementation implements DefaultAPI {
  InlineResponse200 helloWorld() {
    // ... 
  }
}
```

If a new method is added to the spec or an existing method is updated, I would run the generator and it would update `DefaultAPI.java` with any new parameters, return types, API endpoints, etc... And because the implementation (which I create by hand) implements that interface, the build will fail if one of the signatures doesn't correctly match or if an endpoint is not implemented. 

Is the FastAPI OpenAPI Generator supposed to be used in this way? If so, how could I achieve my goals 1 & 2 above with the current output created by the generator? 

### Operating System

macOS

### FastAPI Version

0.85.1

### Python Version

3.9.11

**Comment by sherifnada** [comment]
@iudeen fair enough, I asked the question there as well. I asked here since it wasn't clear to me who owns the OpenAPI Generator templates for FastAPI in that repo, so I thought I'd ask the community here since the ownership is more clearly defined.

**Comment by sherifnada** [comment]
Added an implementation suggestion in this issue https://github.com/OpenAPITools/openapi-generator/issues/13863 would love any feedback or opinions

**Comment by tiangolo** [comment]
FastAPI is actually not intended for you to work on the schema first. The design ideas and intention are more that you would write the code in Python, and that FastAPI makes it so easy to just use standard type annotations and minimal code that it would be easier to maintain that Python code than the spec.

In that case, anything that uses the spec (e.g. docs, generated clients) would use the spec generated by FastAPI.

The code generators would make more sense when you already have a spec and you need to migrate to FastAPI. But all that probably makes less sense if you want to keep updating the schema before the code.
